cli/commons/src/Microsoft.Kiota.Cli.Commons/IO/TableOutputFormatter.cs

```diff
@@ -0,0 +1,257 @@
+    }
+
+    /// <inheritdoc />
+    public async Task WriteOutputAsync(Stream content, IOutputFormatterOptions options, CancellationToken cancellationToken = default) {
+        using var doc = await JsonDocument.ParseAsync(content, cancellationToken: cancellationToken);
+        var table = await ConstructTableAsync(doc, cancellationToken);
+        _ansiConsole.Write(table);
+    }
+
+    /// <summary>
+    /// Asynchronously construct a table given a JSON document
+    /// </summary>
+    /// <param name="document">The parsed json document</param>
+    /// <param name="cancellationToken">The cancellation token</param>
+    public async Task<Table> ConstructTableAsync(JsonDocument document, CancellationToken cancellationToken = default) {
+        var root = GetRootElement(document.RootElement);
+        var firstElement = GetFirstElement(root);
+
+        IEnumerable<string> propertyNames = await GetPropertyNamesAsync(firstElement, cancellationToken);
+        var table = new Table();
+        table.Expand();
+
+        foreach (var propertyName in propertyNames)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            table.AddColumn(propertyName, column =>
+            {
+                if (firstElement.ValueKind == JsonValueKind.Object)
+                {
+                    var hasProp = firstElement.TryGetProperty(propertyName, out var property);
+                    if (property.ValueKind == JsonValueKind.Number)
+                        column.RightAligned().PadLeft(10);
+                }
+            });
+        }
+
+        if (root.ValueKind == JsonValueKind.Array)
+        {
+            foreach (var row in root.EnumerateArray())
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                var rowCols = GetRowColumns(propertyNames, row);
+                table.AddRow(rowCols);
+            }
+        }
+        else if (root.ValueKind == JsonValueKind.Object)
+        {
+            var rowCols = GetRowColumns(propertyNames, root);
+            table.AddRow(rowCols);
+        }
+        else
+        {
+            table.AddRow(GetPropertyValue(root));
+        }
+
+        return table;
+    }
+
+    private static JsonElement GetRootElement(JsonElement input) {
+        var root = input;
+        if (root.ValueKind == JsonValueKind.Object && root.TryGetProperty("value", out var value))
+            root = value;
+        
+        return root;
+    }
+
+    private static JsonElement GetFirstElement(JsonElement root) {
+        var firstElement = root;
+        if (root.ValueKind == JsonValueKind.Array && root.GetArrayLength() > 0)
+        {
+            var enumerated = root.EnumerateArray();
+            firstElement = enumerated.FirstOrDefault();
+        }
+        
+        return firstElement;
+    }
+
+    private static Task<IEnumerable<string>> GetPropertyNamesAsync(JsonElement firstElement, CancellationToken cancellationToken = default) {
```

Comment:
I believe this can be just called synchronously since there is no `async` keyword in it. 
The effect would be removing `GetPropertyNames` in line155 and just calling this method without the optional parameter.

Reply:
I made an async version to allow cancelling the operation using the cancellation token.

Reply:
`GetPropertyNamesAsync` isn't really async though can still be called synchronously. 
I'm not sure if there's no need for both `GetPropertyNamesAsync` and `GetPropertyNames` to exist when they are functionally the same when we could simply call/write one function and ignore passing a parameter when we don't need to use it.

Reply:
It's there to enable the cooperative cancellation based on the cancellation token. Otherwise the function will run to the end when it doesn't need to.